### PR TITLE
BF: Allow to explicitly specify host to aws-s3 resource

### PR DESCRIPTION
**Pull request by yarikoptic**
See first commit (cc7dbd5ac0f2722a6056e1ce4149d535d1bb080a ATM) for details

### CODE REVIEW

**Comment by yarikoptic** [comment]
poor windowses

File: datalad/downloaders/s3.py [hunk 1]
```diff
@@ -46,10 +46,21 @@ class S3Authenticator(Authenticator):
     allows_anonymous = True
     DEFAULT_CREDENTIAL_TYPE = 'aws-s3'
 
-    def __init__(self, *args, **kwargs):
-        super(S3Authenticator, self).__init__(*args, **kwargs)
+    def __init__(self, host=None, **kwargs):
```
Comment (kyleam) [inline datalad/downloaders/s3.py]: Why do the unrelated drop of `*args` here?  While it probably doesn't matter, I don't see any reason to make that backwards-incompatible change.

File: datalad/downloaders/tests/test_s3.py [hunk 2]
```diff
@@ -120,3 +130,31 @@ def deny_access(*args, **kwargs):
     with assert_raises(DownloadError):
         with patch.object(downloader, '_download', deny_access):
             downloader.download("doesn't matter")
+
+
+@with_tempfile
+def test_boto_host_specification(tempfile):
+    # This test relies on a yoh-specific set of credentials to access
```
Comment (kyleam) [inline datalad/downloaders/tests/test_s3.py]: While this of course is useful on your end, my preference would be to not bring this into the main tree.

Comment (yarikoptic) [reply datalad/downloaders/tests/test_s3.py]: where/how to keep it so I don't forget and test whenever I do change anything in related code?

File: datalad/downloaders/s3.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/downloaders/s3.py]: fair enough -- will RF to use `kwargs.pop`. In the longer run -- the whole thing could be RFed to become more explicit/consistent.

Comment (kyleam) [reply datalad/downloaders/s3.py]: > will RF to use `kwargs.pop`

All right, though keeping `*args` around doesn't require doing that.

Comment (yarikoptic) [reply datalad/downloaders/s3.py]: ah, right -- python3 !

File: datalad/downloaders/tests/test_s3.py (hunk 2 above)
Comment (kyleam) [reply datalad/downloaders/tests/test_s3.py]: If having an appropriately named local branch isn't enough, you could leave a one-line comment in this file as a note for yourself.  Anyway, just my two cents.  Feel free to disregard.

File: datalad/downloaders/s3.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/downloaders/s3.py]: ok, this one is done

**Pull request merged**